### PR TITLE
Add visual UI components & login functionality

### PR DESCRIPTION
**Pull request by martinatg**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/UI-DSM/pulls) open
- [x] I have verified that I am following the UI-DSM [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/UI-DSM/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

This PR add mostly styling changes and prepares UI components for future development. It includes:

- CSS vars
- Layout components
- Button
- Input
- Login styling
- Login route guard on MainLayout

### CODE REVIEW

File: UI_DSM/UI_DSM.Client/Components/App/AppButton/AppButton.razor [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<!--------------------------------------------------------------------------------------------------------
+// ConfirmCancelPopup.razor
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Components/App/AppButton/AppButton.razor:2]: Please change the name of the file on the header

File: UI_DSM/UI_DSM.Client/Components/App/AppButton/AppButton.razor [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        Primary,
+        Secondary,
+        Success,
+        Tertiary
+    }
+
+    [Parameter] public VariantValue Variant { get; set; } = VariantValue.Primary;
+    private string buttonVariant { get; set; }
+
+    protected override void OnInitialized()
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Components/App/AppButton/AppButton.razor:33]: Please also comment methods and all properties

File: UI_DSM/UI_DSM.Client/Components/App/AppInput/AppInput.razor [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    <input class="app-input__input @CssClass" @bind="@CurrentValue" type="@Type" />
+
+  @if (!string.IsNullOrWhiteSpace(Note))
+  {
+    <span class="app-input__note">@Note</span>
+  }
+</label>
+
+@code {
+    [Parameter] public string Id { get; set; }
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Components/App/AppInput/AppInput.razor:30]: Comments + Header file name

File: UI_DSM/UI_DSM.Client/Components/App/SearchBar/SearchBar.razor [hunk 4]
```diff
@@ -1,20 +1,20 @@
-﻿<!--------------------------------------------------------------------------------------------------------
-// Header.razor
+<!--------------------------------------------------------------------------------------------------------
+// ConfirmCancelPopup.razor
 // Copyright (c) 2022 RHEA System S.A.
 // 
-// Author: Antoine Théate, Sam Gerené, Alex Vorobiev, Alexander van Delft
+// Author: Antoine Théate, Sam Gerené, Alex Vorobiev, Alexander van Delft, Martin Risseeuw
 // 
 // This file is part of UI-DSM.
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Components/App/SearchBar/SearchBar.razor:7]: Header file name

File: UI_DSM/UI_DSM.Client/Components/App/TaskItem/TaskItem.razor [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<!--------------------------------------------------------------------------------------------------------
+// ConfirmCancelPopup.razor
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Components/App/TaskItem/TaskItem.razor:2]: header file name

File: UI_DSM/UI_DSM.Client/Components/App/TaskItem/TaskItem.razor [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    @if (isDone) {
+      <FeatherCheckCircle Size="24" Color="#1FAB77" StrokeWidth="2" />
+    }
+  </div>
+  <div class="task-item__body">
+    @TaskDescription
+  </div>
+</div>
+
+@code {
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Components/App/TaskItem/TaskItem.razor:27]: Comments

File: UI_DSM/UI_DSM.Client/Components/App/UserAvatar/UserAvatar.razor [hunk 7]
```diff
@@ -0,0 +1,33 @@
+<!--------------------------------------------------------------------------------------------------------
+// ConfirmCancelPopup.razor
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Components/App/UserAvatar/UserAvatar.razor:2]: Header file name

File: UI_DSM/UI_DSM.Client/Components/App/UserAvatar/UserAvatar.razor [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    <img class="user-avatar__image" src="@UserImage" />
+  }
+  else
+  {
+    <span class="user-avatar__initials">@UserInitials</span>
+  }
+</figure>
+
+@code {
+  [Parameter]
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Components/App/UserAvatar/UserAvatar.razor:26]: Comments

File: UI_DSM/UI_DSM.Client/Pages/LoginPage/LoginPage.razor [hunk 9]
```diff
@@ -0,0 +1,33 @@
+<!--------------------------------------------------------------------------------------------------------
+// Index.razor
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Pages/LoginPage/LoginPage.razor:2]: Header file name

File: UI_DSM/UI_DSM.Client/Pages/ReviewPage/ReviewPage.razor [hunk 10]
```diff
@@ -0,0 +1,26 @@
+<!--------------------------------------------------------------------------------------------------------
+// Index.razor
+// Copyright (c) 2022 RHEA System S.A.
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Pages/ReviewPage/ReviewPage.razor:3]: Header file name

File: UI_DSM/UI_DSM.Client/Shared/CleanLayout/CleanLayout.razor [hunk 11]
```diff
@@ -1,4 +1,4 @@
-﻿<!--------------------------------------------------------------------------------------------------------
+<!--------------------------------------------------------------------------------------------------------
 // MainLayout.razor
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Shared/CleanLayout/CleanLayout.razor:2]: Header file name

File: UI_DSM/UI_DSM.Client/Shared/MainLayout/MainLayout.razor [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+        </div>
+        
+        <aside  class="default-layout__sidebar default-layout__sidebar__right">
+            <div class="sidebar__header">Comments</div>
+        </aside>
+    </div>
+</div>
+
+@code {
+	@inject NavigationManager NavigationManager
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Shared/MainLayout/MainLayout.razor:48]: Comments

File: UI_DSM/UI_DSM.Client/Shared/TopMenu/TopMenu.razor [hunk 13]
```diff
@@ -0,0 +1,94 @@
+<!--------------------------------------------------------------------------------------------------------
+// ConfirmCancelPopup.razor
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Shared/TopMenu/TopMenu.razor:2]: Header file name

File: UI_DSM/UI_DSM.Client/Shared/TopMenu/TopMenu.razor [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+          </ul>
+
+        </DxDropDown>
+      </div>
+    </div>
+  </div>
+</nav>
+
+@code {
+  bool IsUserMenuOpen { get; set; } = false;
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/Shared/TopMenu/TopMenu.razor:93]: Comments

File: UI_DSM/UI_DSM.Client/ViewModels/Components/ILoginViewModel.cs [hunk 15]
```diff
@@ -43,5 +43,7 @@ public interface ILoginViewModel
         /// </summary>
         /// <returns>A <see cref="Task" /></returns>
         Task ExecuteLogin();
+
+        void NavigateIfLoggedIn();
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/ViewModels/Components/ILoginViewModel.cs:47]: Comments

File: UI_DSM/UI_DSM.Client/ViewModels/Components/LoginViewModel.cs [hunk 16]
```diff
@@ -85,10 +94,20 @@ public async Task ExecuteLogin()
...
+                this.NavigateIfLoggedIn();
+            }
+
             if (response.Errors != null && response.Errors.Any())
             {
                 this.ErrorMessage = string.Join("\n", response.Errors);
             }
         }
+
+        public void NavigateIfLoggedIn()
```
Comment (antoineatrhea) [inline UI_DSM/UI_DSM.Client/ViewModels/Components/LoginViewModel.cs:108]: Comments (same as the ILoginViewModel.cs file)

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
Header file names needs to be renamed and small documentation still missing.

**Comment by martinatg** [comment]
@antoineatrhea I updated the comments & file headers.

**Review by antoineatrhea — APPROVED** [review]

**Pull request merged**